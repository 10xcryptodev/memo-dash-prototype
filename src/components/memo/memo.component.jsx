import React, { Component } from 'react'
import { Comment } from 'semantic-ui-react'
import MemoActionsContainer from './actions/memo-actions.container'
import MemoAvatarContainer from './avatar/memo-avatar.container'
import MemoContentContainer from './content/memo-content.container'
import './memo.styles.css'
import MemoContainer from './memo.container'

export default class MemoComponent extends Component {
  componentDidMount() {
    const { memo, showReplies, onLoadReplies } = this.props
    if (showReplies && memo.memoRepliesCount > 0) {
      onLoadReplies(memo)
    }
  }

  render() {
    const { memo, showReplies, replies, openModalOnClick, onModalOpenClicked } = this.props
    return (
      <Comment onClick={!!openModalOnClick ? () => onModalOpenClicked(memo) : undefined}>
        <MemoAvatarContainer memo={memo} />
        <Comment.Content>
          <MemoContentContainer memo={memo} />
          <MemoActionsContainer memo={memo} />
        </Comment.Content>
        {showReplies &&
          !!replies && (
            <Comment.Group>
              {replies.map(reply => <MemoContainer showReplies={true} memo={reply} key={reply.idx} />)}
            </Comment.Group>
          )}
      </Comment>
    )
  }
}
