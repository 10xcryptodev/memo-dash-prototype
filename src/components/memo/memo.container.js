import { connect } from 'react-redux'

import MemoComponent from './memo.component'
import { openMemoModal, getMemoReplies, closeMemoModal } from '../../store/actions'
import { getMemosByCombinedIds, isMemoOfCurrentUser } from '../../store/selectors'

const mapStateToProps = (state, ownProps) => {
  const { memo, showReplies } = ownProps

  if (!memo) return {}

  let replies = undefined
  if (showReplies) {
    replies = getMemosByCombinedIds(memo.replyIds)(state)
  }

  return { replies, showDelete: isMemoOfCurrentUser(memo)(state) }
}

const mapDispatchToProps = dispatch => {
  return {
    onModalOpenClicked: memo => dispatch(openMemoModal(memo)),
    closeModal: () => dispatch(closeMemoModal()),
    onLoadReplies: memo => dispatch(getMemoReplies(memo.username, memo.idx))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MemoComponent)
