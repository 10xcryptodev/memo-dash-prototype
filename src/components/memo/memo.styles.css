.memo {
  cursor: pointer;
}

.memo .actions > span {
  margin: 0 0.75em 0 0;
}
