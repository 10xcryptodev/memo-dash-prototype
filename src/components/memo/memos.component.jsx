import React from 'react'
import MemoContainer from './memo.container'
import { Comment, Segment } from 'semantic-ui-react'
import MemoModalContainer from './modal/memo-modal.container'

const MemosComponent = props => {
  const { memos } = props

  if(!memos){
    return null;
  }

  return (
    <Segment basic center>
      {memos ? (
        <Comment.Group>
          {Object.keys(memos)
            .sort((a, b) => a.memoDatetime < b.memoDatetime)
            .map(memoId => (
              <MemoContainer
                key={memoId}
                showBorders={true}
                openModalOnClick={true}
                showReplies={false}
                memo={memos[memoId]}
              />
            ))}
          <MemoModalContainer />
        </Comment.Group>
      ) : (
        'No memos available'
      )}
    </Segment>
  )
}

export default MemosComponent
