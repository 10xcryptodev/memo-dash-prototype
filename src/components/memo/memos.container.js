import { connect } from 'react-redux'

import MemosComponent from './memos.component'

const mapStateToProps = (state, ownProps) => {
  return {}
}

const mapDispatchToProps = dispatch => {
  return {}
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MemosComponent)
