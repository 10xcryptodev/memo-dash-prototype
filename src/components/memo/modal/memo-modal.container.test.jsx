import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import MemoModalContainer from './memo-modal.container'
import { combineMemoId } from '../../../store/reducers/memo.reducer'

import testUsers from '../../../test-utils/test-users'

describe('<MemoModalContainer />', () => {
  let store
  const alice = testUsers['alice']
  const bob = testUsers['bob']
  let mockStore

  beforeEach(() => {
    mockStore = configureStore()

    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  describe('Shallow rendering', () => {
    describe('should render', () => {
      it('without crashing', () => {
        store = mockStore({
          memoModal: { opened: false },
          memo: { memos: [] }
        })

        const wrapper = shallow(<MemoModalContainer memo={alice.memos[0]} />, { context: { store } })
        expect(wrapper).toMatchSnapshot()
      })

      it('opened modal', () => {
        const memo = alice.memos[0]
        const combinedMemoId = combineMemoId(memo.username, memo.idx)
        store = mockStore({
          memoModal: { opened: true, openedMemo: combinedMemoId },
          memo: { memos: { [combinedMemoId]: memo } }
        })

        const wrapper = shallow(<MemoModalContainer memo={alice.memos[0]} />, { context: { store } })
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
