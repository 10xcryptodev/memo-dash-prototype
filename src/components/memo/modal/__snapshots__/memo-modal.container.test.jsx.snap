// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoModalContainer /> Shallow rendering should render opened modal 1`] = `
<MemoModalComponent
  memo={
    Object {
      "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
      "idx": 1,
      "memoDatetime": "2018-07-26T16:44:56.000Z",
      "memoLikesCount": 1,
      "memoRepliesCount": 0,
      "memoText": "Suscipit vel nobis.",
      "memoTipTotal": 0,
      "username": "alice",
    }
  }
  onModalCloseClicked={[Function]}
  opened={true}
/>
`;

exports[`<MemoModalContainer /> Shallow rendering should render without crashing 1`] = `
<MemoModalComponent
  onModalCloseClicked={[Function]}
  opened={false}
/>
`;
